broker-server/src/test/java/org/aktin/broker/TestBroker.java

```diff
@@ -436,4 +484,102 @@ public void filterRequestByXPath() throws IOException{
 		assertEquals(qid, list.get(0).getId());
 	
 	}
+	
+	@Test
+	public void getNonexistingBrokerRequest() throws IOException {
+		BrokerAdmin a = initializeAdmin();
+		assertNull(a.getRequestDefinition(99, "text/x-test-2"));
+	}
+	
+	@Test
+	public void addRequestDefinitionToNonexistingRequest() throws IOException {
+		BrokerAdmin a = initializeAdmin();
+		a.putRequestDefinition(999, "text/x-test-1", "test1");
+	}
+	
+	@Test
+	public void deleteNonexistingBrokerRequest() throws IOException {
+		BrokerAdmin a = initializeAdmin();
+		a.deleteRequest(999);
+	}
+	
+	@Test
+	public void addTargetNodesToNonexistingRequest() throws IOException {
+		BrokerAdmin a = initializeAdmin();
+		a.setRequestTargetNodes(999, new int[]{1});
+	}
+	
+	@Test
+	public void getTargetNodesOfNonexistingRequest() throws IOException {
+		BrokerAdmin a = initializeAdmin();
+		assertNull(a.getRequestTargetNodes(999));
+	}
+	
+	@Test
+	public void getTargetNodesOfUntargetedRequest() throws IOException {
```

Comment:
This test is in conflict with the added javadoc at BrokerAdmin#getRequestTargetNodes (L158) saying that an exception is thrown for untargeted requests